codes/environment/pos_env.py

```
@@ -31,66 +39,82 @@ def __init__(self, config, aligned_df: pd.DataFrame):
-        Apply action for a single validator and return (next_state, reward, done, info)
-        Reward is shaped by uptime, missed metrics, stake and whether slashed.
+        Public API to update agent state. Only use this method to mutate validator state externally.
+        Example: env.set_agent_state(vid, uptime=0.9, slashed=True)
         """
         st = self._state[validator_id]
-        name = ACTIONS.get(action, "abstain")
+        for k, v in kwargs.items():
+            if hasattr(st, k):
+                setattr(st, k, v)
+            else:
+                raise AttributeError(f"EnvState has no attribute '{k}'")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)